panels/user-accounts/um-app-permissions.c

```diff
@@ -0,0 +1,787 @@
+
+      if (age > maximum_age)
+        maximum_age = age;
+    }
+
+  g_debug ("Effective age for this user: %u", maximum_age);
+
+  rating_system = get_content_rating_system (self->user);
+  rating_age_category = gs_utils_content_rating_age_to_str (rating_system, maximum_age);
+
+  gtk_button_set_label (self->restriction_button, rating_age_category);
+}
+
+static void
+update_allow_app_installation (UmAppPermissions *self)
+{
+  gboolean allow_system_installation;
+  gboolean allow_user_installation;
+
+  allow_system_installation = epc_app_filter_is_system_installation_allowed (self->filter);
+  allow_user_installation = epc_app_filter_is_user_installation_allowed (self->filter);
+
+  g_signal_handlers_block_by_func (self->allow_system_installation_switch,
+                                   on_allow_installation_switch_active_changed_cb,
+                                   self);
+
+  g_signal_handlers_block_by_func (self->allow_user_installation_switch,
+                                   on_allow_installation_switch_active_changed_cb,
+                                   self);
+
+  gtk_switch_set_active (self->allow_system_installation_switch, allow_system_installation);
+  gtk_switch_set_active (self->allow_user_installation_switch, allow_user_installation);
+
+  g_debug ("Allow system installation: %s", allow_system_installation ? "yes" : "no");
+  g_debug ("Allow user installation: %s", allow_user_installation ? "yes" : "no");
+
+  g_signal_handlers_unblock_by_func (self->allow_system_installation_switch,
+                                     on_allow_installation_switch_active_changed_cb,
+                                     self);
+
+  g_signal_handlers_unblock_by_func (self->allow_user_installation_switch,
+                                     on_allow_installation_switch_active_changed_cb,
+                                     self);
+}
+
+static void
+setup_parental_control_settings (UmAppPermissions *self)
+{
+  gtk_widget_set_visible (GTK_WIDGET (self), self->filter != NULL);
+
+  if (!self->filter)
+    return;
+
+  g_hash_table_remove_all (self->blacklisted_apps);
+
+  update_oars_level (self);
+  update_categories_from_language (self);
+  update_allow_app_installation (self);
+  reload_apps (self);
+}
+
+static gchar*
+get_flatpak_ref_for_app_id (UmAppPermissions *self,
+                            const gchar      *flatpak_id)
+{
+  g_autoptr(FlatpakInstalledRef) ref = NULL;
+  g_autoptr(GError) error = NULL;
+
+  g_assert (self->system_installation != NULL);
+  g_assert (self->user_installation != NULL);
+
+  ref = flatpak_installation_get_current_installed_app (self->user_installation,
+                                                        flatpak_id,
+                                                        self->cancellable,
+                                                        &error);
+
+  if (error && !g_error_matches (error, FLATPAK_ERROR, FLATPAK_ERROR_NOT_INSTALLED))
```

Comment:
In the case that `(error != NULL && g_error_matches (error, FLATPAK_ERROR, FLATPAK_ERROR_NOT_INSTALLED)`, this branch will not be taken and `error` will remain set through to the `flatpak_installation_get_current_installed_app()` call below. Passing a non-`NULL` `GError` into a call is disallowed.